### PR TITLE
[ADD] l10n_fr_pos_cert: ensure anti-fraud of POS sales data

### PR DESCRIPTION
**Pull request by smetl**
The generation of such archives is a french law requirements.

task: https://www.odoo.com/web#id=47989&view_type=form&model=project.task&action=333&active_id=967&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_fr_pos_cert/data/pos_inalterability.xml [hunk 1]
```diff
@@ -14,4 +14,11 @@ action = env['pos.order']._check_hash_integrity(env.user.company_id.id)
     </record>
 
     <menuitem action="action_check_hash_integrity_pos_orders" id="menu_check_pos_order_integrity" parent="pos_fr_statements_menu" sequence="90"/>
-</odoo>
\ No newline at end of file
+
+    <record id="archive_posted" model="mail.message.subtype">
```
Comment (qdp-odoo) [inline addons/l10n_fr_pos_cert/data/pos_inalterability.xml]: to remove

File: addons/l10n_fr_pos_cert/models/res_company.py [hunk 2]
```diff
@@ -25,4 +33,171 @@ def write(self, vals):
...
+        # Filename
+        filename = 'archive_journal_items.csv'
+
+        # Fields
+        fields = ['id', 'name', 'quantity', 'product_uom_id', 'product_id', 'debit', 'credit', 'balance',
+                  'debit_cash_basis', 'credit_cash_basis', 'balance_cash_basis', 'amount_currency',
+                  'company_currency_id', 'currency_id', 'amount_residual', 'amount_residual_currency',
+                  'tax_base_amount', 'account_id', 'move_id', 'ref', 'payment_id', 'statement_line_id',
+                  'statement_id', 'reconciled', 'full_reconcile_id', 'date_maturity', 'date', 'tax_line_id',
+                  'analytic_account_id', 'partner_id', 'user_type_id', 'tax_exigible', 'create_date']
```
Comment (qdp-odoo) [inline addons/l10n_fr_pos_cert/models/res_company.py]: ask fgi for the right columns to archive

File: addons/l10n_fr_pos_cert/models/res_company.py [hunk 3]
```diff
@@ -25,4 +33,171 @@ def write(self, vals):
...
+
+        :return: filename, csv_content
+        '''
+        self.ensure_one()
+
+        # Filename
+        filename = 'archive_post_order_lines.csv'
+
+        # Fields
+        fields = ['id', 'name', 'notice', 'product_id', 'price_unit', 'qty', 'discount', 'order_id', 'create_date']
```
Comment (qdp-odoo) [inline addons/l10n_fr_pos_cert/models/res_company.py]: see with fgi

**Comment by smetl** [comment]
Specs changed.

**Pull request closed without merging**